src/osd/OSD.cc

```diff
@@ -10075,6 +10094,60 @@ void OSD::handle_conf_change(const ConfigProxy& conf,
   }
 }
 
+void OSD::maybe_override_max_osd_capacity_for_qos()
+{
+  // If the scheduler enabled is mclock, override the default
+  // osd capacity with the value obtained from running the
+  // osd bench test. This is later used to setup mclock.
+  if (cct->_conf.get_val<std::string>("osd_op_queue") == "mclock_scheduler") {
+    // Write 200 4MiB objects with blocksize 4KiB
+    int64_t count = 12288000; // Count of bytes to write
+    int64_t bsize = 4096;     // Block size
+    int64_t osize = 4194304;  // Object size
+    int64_t onum = 200;       // Count of objects to write
+    double elapsed = 0.0;     // Time taken to complete the test
+    stringstream ss;
+    dout(1) << __func__
+            << " osd bench test -"
+            << " max_bytes_count: " << count
+            << " blocksize: " << bsize
+            << " max_object_count: " << onum
+            << dendl;
+    int ret = run_osd_bench_test(count, bsize, osize, onum, &elapsed, ss);
+    if (ret != 0) {
+      dout(1) << __func__
+              << " osd bench err: " << ret
+              << " osd bench errstr: " << ss.str()
+              << dendl;
+      dout(1) << " Using default osd_mclock_max_capacity_iops" << dendl;
+    } else {
+      double rate = count / elapsed;
+      double iops = rate / bsize;
+      dout(1) << __func__
+              << " osd bench result -"
+              << std::fixed << std::setprecision(3)
+              << " bandwidth (MiB/sec): " << rate / (1024 * 1024)
+              << " iops: " << iops
+              << " elapsed_sec: " << elapsed
+              << dendl;
+
+      // Override the appropriate config option
+      if (store_is_rotational) {
+        cct->_conf.set_val(
+          "osd_mclock_max_capacity_iops_hdd", std::to_string(iops));
```

Comment:
mclock_max_capacity_iops currently overrides the hdd or ssd settings we're using here. since this is a direct measurement of the device, I'm not sure there's much use in having the generic setting - and it could cause confusion as we're not setting it here (and if we did we'd be setting two settings).

what do you think about getting rid of the generic setting, and only having the hdd/ssd versions?
alternatively, we could set only the generic setting here, since it takes precedence.

Reply:
For this option I agree that we can remove the generic option and just have the hdd and ssd versions. In case the test fails for some reason, we can fall back on the default values for hdd and ssd. Please verify.

Reply:
sounds good to me